src/Language/DifferentialDatalog/Compile.hs

```diff
@@ -794,51 +765,51 @@ mkValueFromRecord d@DatalogProgram{..} =
     mkIdxId2NameC                                                                                   $$
     mkIdxIdMap d                                                                                    $$
     mkIdxIdMapC d                                                                                   $$
-    "pub fn relval_from_record(rel: Relations, _rec: &record::Record) -> Result<Value, String> {"   $$
+    "pub fn relval_from_record(rel: Relations, _rec: &record::Record) -> result::Result<DDValue, String> {" $$
     "    match rel {"                                                                               $$
-    (nest' $ nest' $ vcat $ punctuate comma entries)                                                $$
+    (nest' $ nest' $ vcommaSep entries)                                                             $$
     "    }"                                                                                         $$
     "}"                                                                                             $$
-    "pub fn relkey_from_record(rel: Relations, _rec: &record::Record) -> Result<Value, String> {"   $$
+    "pub fn relkey_from_record(rel: Relations, _rec: &record::Record) -> result::Result<DDValue, String> {" $$
     "    match rel {"                                                                               $$
-    (nest' $ nest' $ vcat $ key_entries)                                                            $$
+    (nest' $ nest' $ vcommaSep key_entries)                                                         $$
```

Comment:
was this a bug? these do not seem to be equivalent.

Reply:
I think the comma used to be in `key_entries`.